pkg/ir/deps/sql_program_deps_graph.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2020 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deps
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"sqlflow.org/sqlflow/pkg/parser"
+)
+
+// Statement represents a graph node of one statement.
+type Statement struct {
+	Statement string
+	// index of the statement in the SQL program
+	Order int
+	// Statement's input/output must be a table.
+	Inputs  []*Table
+	Outputs []*Table
+}
+
+// TableType can be table or model
+type TableType string
```

Comment:
Why not type int here, but a string?

Reply:
TableType will be used to generate a unique table name like table.my_table or model.my_model when constructing the graph and solving hazards.